### PR TITLE
feat(fs): improved API

### PR DESCRIPTION
**Pull request by amrbashir**

### CODE REVIEW

**Comment by FabianLars** [comment]
It doesn't compile with the watch feature enabled. Just a fairly small change though (the main thing is that resolve_path expects a SafePathBuf) but i won't push it to not risk conflicts if you have local commits left :)

(p.s. opened https://github.com/tauri-apps/plugins-workspace/pull/761 to test more features in ci)

The main api change looks solid to me though (though i do think node's fs api is a bit weird idk), and what i tested so far worked nicely! 🥳 (i'll do some more testing tomorrow)
I do however think that this PR should have 2 approvals (maybe i just don't trust myself enough lol).

**Comment by amrbashir** [comment]
Alright, watcher flag should be fine now, and feel free to push your changes if you need, I don't have any more changes planned.

> though i do think node's fs api is a bit weird idk)

The API is inspired from Deno and Node and the main goal was to make it more familiar to those coming from node.

File: plugins/fs/src/commands.rs
Comment (lucasfernog) [inline plugins/fs/src/commands.rs]: shouldn't we just set the mode here?

File: plugins/fs/src/commands.rs
Comment (lucasfernog) [inline plugins/fs/src/commands.rs]: I think they should all default to false (via serde instead of all the logic here); what if I want a file to be write-only for security reasons?

**Comment by FabianLars** [comment]
Why was the `close` export removed? Isn't that required for `open` to make sense? (honestly already forgot everything about this pr so genuine question)

**Comment by amrbashir** [comment]
I guess for security reasons, since `close` accepts any `rid`, I think they should use `.close()` on the returned `FsFile` class

File: plugins/fs/src/commands.rs
Comment (amrbashir) [reply plugins/fs/src/commands.rs]: tbh I don't know, this was copied from deno implementation but I guess they have a default mode or something, I will just change that to use the mode directly.

File: plugins/fs/src/commands.rs
Comment (amrbashir) [reply plugins/fs/src/commands.rs]: sure

Comment (amrbashir) [reply plugins/fs/src/commands.rs]: should I also change the default (the `None` case) behavior when options were not provided, to set `read` to `false`?

Comment (amrbashir) [reply plugins/fs/src/commands.rs]: actually we need to keep `read` to `true`, otherwise opening a file without `read` or `write` will fail.

Comment (amrbashir) [reply plugins/fs/src/commands.rs]: if someone wants a write-only file, they will need to specify `write: true, read: false`

**Comment by amrbashir** [comment]
btw this PR is blocked until https://github.com/tauri-apps/tauri/pull/8370

**Review by lucasfernog — DISMISSED** [review]

**Review by lucasfernog — APPROVED** [review]

**Pull request merged**
